code-challenges.test.js

```diff
@@ -1,73 +1,161 @@
+})
 
-// --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+
+// Create a function 
+//that takes in an array 
+//and returns a new array 
+        //of only odd numbers 
+            //sorted from least to greatest.
 var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
-// Expected output: [-9, 7, 9, 199]
+const sortedOddNums =(array) =>{
+    //filter() new variable filteredArray =array.filter() look up filter syntax
+    let filteredArray = array.filter((value , index) => {
+           if(typeof value === "number" && value % 2 !== 0){
+        return value
+        }
+    })
+    //sort() new  variable sortedFilterdArray = filteredArray.sort() look up sort syntax
+    let sortedFilterdArray = filteredArray.sort((a,b)=> a-b)
 
-var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
-// Expected output: [-823, 7, 23]
+    //return sortedFilterdArray 
+return sortedFilterdArray;
+}
+console.log(sortedOddNums(fullArr1));
 
+// // Expected output: [-9, 7, 9, 199]
 
+ var  fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+// // Expected output: [-823, 7, 23]
 
 
-// b) Create the function that makes the test pass.
 
 
+// // b) Create the function that makes the test pass.
 
 
 
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
-// a) Create a test with expect statements for each of the variables provided.
 
-var numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
+// // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
-var numbersToAdd2 = [0, 7, -8, 12]
-// Expected output: [0, 7, -1, 11]
+// // a) Create a test with expect statements for each of the variables provided.
+describe("sumOfArray ", () =>{
 
-var numbersToAdd3 = []
-// Expected output: []
+    it("returns only sorted odd numbers", () =>{
+        expect(sumOfArray ( numbersToAdd1)).toEqual([2, 4, 45, 9])
+        expect(sumOfArray ( numbersToAdd2)).toEqual([0, 7, -8, 12])
+        expect(sumOfArray ( numbersToAdd3)).toEqual([])
+    })
+})
+ var numbersToAdd1 = [2, 4, 45, 9]
+// // Excpected output: [2, 6, 51, 60]
 
+ var numbersToAdd2 = [0, 7, -8, 12]
+// // Expected output: [0, 7, -1, 11]
 
+ var numbersToAdd3 = []
+// // Expected output: []
 
 
 
-// b) Create the function that makes the test pass.
+//Create a function called sumOfArray
+   //that takes in an array witch means parameter (array)
+        //and returns an array of the accumulating sum. 
+            //An empty array should return an empty array.
+            //I found a method called .reduce() but the issue is that it is giving me the sum of all the numbers
```

Comment:
So a key life lesson here: don't use a complicated tool where a simple tool will do just fine. .reduce() is only useful if we can make it work. What are some other tools that are easier to implement that can do the same thing?